api/envoy/extensions/filters/http/aws_request_signing/v3/aws_request_signing.proto

```diff
@@ -63,6 +74,11 @@ message AwsRequestSigning {
   // - exact: bar
   // When applied, all headers that start with "x-envoy" and headers "foo" and "bar" will not be signed.
   repeated type.matcher.v3.StringMatcher match_excluded_headers = 5;
+
+  // Optional Signing algorithm specifier, either aws_sigv4 or aws_sigv4a, defaulting to aws_sigv4.
+  // When using aws_sigv4a, the region specifier is now used as a region set. A region set is a comma seperated list of
+  // AWS regions, such as ``us-east-1,us-east-2`` or wildcard ``*``
+  SigningAlgorithm signing_algorithm = 6;
```

Comment:
so the way it should work is 
- sigv4 supports a region specifier only
- sigv4a supports a single region specifier or multiple regions comma seperated, of which either of these can contain wildcards

it's not possible to determine the intention behind the signing just due to the region string, because a single region (eg ap-southeast-2) is valid for either sigv4 or sigv4a. 

I think the best approach is to be explicit, in that you can either specify `signing_algorithm: aws_sigv4` and optionally `region: <single region>`, or you specify `signing_algorithm: aws_sigv4a` and optionally `region_set: <one or more regions with optional wildcards>`. However, that likely changes the config format, as region or region set would be a sub specifier of signing_algorithm.
What are your thoughts here?

Reply:
@mattklein123 interested in your thoughts here. My current implementation overloads region - if signing algorithm is sigv4a then the contents of region is used as a region set. No real impact to existing users in this model, as sigv4a does not yet exist. But it's not as explicit as it could be.
I'm not really sure of the 'right' way to make a dependent parameter in the protobuf either, so if I did go the explicit route of creating a dedicated region_set parameter I'd need to find an example of how to bind the parameters together.

Reply:
IMO, explicit switch would be better and clear for end users. Implicitly behavior sometimes will make debugging a nightmare.

Reply:
ok, have converted to an explicit parameter

Reply:
Maybe you mis-understood my comment. I agree with your design here, you needn't additional change here. 🤣